File: sdk/java/sgraal-java/src/main/java/com/sgraal/PreflightRequest.java
package com.sgraal;
import java.util.List;

public class PreflightRequest {
    public List<MemoryEntry> memoryState;
    public String domain;
    public String actionType;

    public PreflightRequest(List<MemoryEntry> memoryState, String domain, String actionType) {
        this.memoryState = memoryState; this.domain = domain; this.actionType = actionType;
    }
}

File: sdk/dotnet/Sgraal/Models/PreflightRequest.cs
namespace Sgraal.Models;
using System.Collections.Generic;
public class PreflightRequest
{
    public List<MemoryEntry> MemoryState { get; set; } = new();
    public string Domain { get; set; } = "general";
    public string ActionType { get; set; } = "reversible";
}
